kubeflow/tf-serving/prototypes/tf-serving.jsonnet

```diff
@@ -0,0 +1,240 @@
+  s3SecretSecretaccesskeyKeyName: "AWS_SECRET_ACCESS_KEY",
+
+  // S3 region
+  s3AwsRegion: "us-west-1",
+
+  // TODO(jlewi): We should use util.toBool to automatically conver to actual boolean values.
+  // The use of strings is left over from when they were prototype parameters which only supports string type.
+  // true Whether or not to use https for S3 connections
+  s3UseHttps: "true",
+
+  // Whether or not to verify https certificates for S3 connections
+  s3VerifySsl: "true",
+
+  // URL for your s3-compatible endpoint.
+  s3Endpoint: "http://s3.us-west-1.amazonaws.com,",
+} + params;
+
+local s3Env = [
+  { name: "AWS_ACCESS_KEY_ID", valueFrom: { secretKeyRef: { name: s3params.s3SecretName, key: s3params.s3SecretAccesskeyidKeyName } } },
+  { name: "AWS_SECRET_ACCESS_KEY", valueFrom: { secretKeyRef: { name: s3params.s3SecretName, key: s3params.s3SecretSecretaccesskeyKeyName } } },
+  { name: "AWS_REGION", value: s3params.s3AwsRegion },
+  { name: "S3_REGION", value: s3params.s3AwsRegion },
+  { name: "S3_USE_HTTPS", value: s3params.s3UseHttps },
+  { name: "S3_VERIFY_SSL", value: s3params.s3VerifySsl },
+  { name: "S3_ENDPOINT", value: s3params.s3Endpoint },
+];
+
+// Optional features.
+// TODO(lunkai): Add Istio
+// TODO(lunkai): Add request logging
+
+local service = {
+  apiVersion: "v1",
+  kind: "Service",
+  metadata: {
+    labels: {
+      app: appName,
+    },
+    name: appName,
+    namespace: namespace,
+    annotations: {
+      "getambassador.io/config":
+        std.join("\n", [
+          "---",
+          "apiVersion: ambassador/v0",
+          "kind:  Mapping",
+          "name: tfserving-mapping-" + name + "-get",
+          "prefix: /models/" + name + "/",
+          "rewrite: /",
+          "method: GET",
+          "service: " + name + "." + namespace + ":8000",
+          "---",
+          "apiVersion: ambassador/v0",
+          "kind:  Mapping",
+          "name: tfserving-mapping-" + name + "-post",
+          "prefix: /models/" + name + "/",
+          "rewrite: /model/" + name + ":predict",
+          "method: POST",
+          "service: " + name + "." + namespace + ":8000",
+        ]),
+    },  //annotations
+  },
+  spec: {
+    ports: [
+      {
+        name: "grpc-tf-serving",
+        port: 9000,
+        targetPort: 9000,
+      },
+      {
+        name: "http-tf-serving-proxy",
+        port: 8000,
+        targetPort: 8000,
+      },
+    ],
+    selector: {
+      app: appName,
+    },
+    type: "ClusterIP",
+  },
+};  // service
+
+local modelServerContainer = {
+  command: [
+    "/usr/bin/tensorflow_model_server",
+  ],
+  args: [
+    "--port=9000",
+    "--model_name=" + params.modelName,
+    "--model_base_path=" + params.modelBasePath,
+  ],
+  image: modelServerImage,
+  imagePullPolicy: "IfNotPresent",
+  name: name,
+  ports: [
+    {
+      containerPort: 9000,
+    },
+  ],
+  env: []
+       + if util.toBool(params.s3Enable) then s3Env else []
```

Comment:
I would move this into deployment.mapContainers  in the kubeflow/tf-serving/prototypes/tf-serving-aws.jsonnet file as described above